src/netplan.c

```diff
@@ -203,6 +203,66 @@ write_bond_params(yaml_event_t* event, yaml_emitter_t* emitter, const NetplanNet
 err_path: return FALSE; // LCOV_EXCL_LINE
 }
 
+static gboolean
+write_vxlan_params(yaml_event_t* event, yaml_emitter_t* emitter, const NetplanNetDefinition* def)
+{
+    if (DIRTY(def, def->vxlan_params)
+        || def->vxlan_params.remote
+        || def->vxlan_params.local
+        || def->vxlan_params.group
+        || def->vxlan_params.tos
+        || def->vxlan_params.ttl
+        || def->vxlan_params.mac_learning
+        || def->vxlan_params.fdb_ageing
+        || def->vxlan_params.max_fdb_entries
+        || def->vxlan_params.reduce_arp_proxy
+        || def->vxlan_params.l2_miss_notification
+        || def->vxlan_params.l3_miss_notification
+        || def->vxlan_params.route_short_circuit
+        || def->vxlan_params.udp_checksum
+        || def->vxlan_params.udp6_zero_checksum_tx
+        || def->vxlan_params.udp6_zero_checksum_rx
+        || def->vxlan_params.remote_checksum_tx
+        || def->vxlan_params.remote_checksum_rx
+        || def->vxlan_params.group_policy_extension
+        || def->vxlan_params.generic_protocol_extension
+        || def->vxlan_params.destination_port
+        || def->vxlan_params.port_range
+        || def->vxlan_params.flow_label
+        || def->vxlan_params.ip_do_not_fragment
+        || def->vxlan_params.independent) {
+        YAML_SCALAR_PLAIN(event, emitter, "parameters");
+        YAML_MAPPING_OPEN(event, emitter);
+        YAML_STRING(def, event, emitter, "remote", def->vxlan_params.remote);
+        YAML_STRING(def, event, emitter, "local", def->vxlan_params.local);
+        YAML_STRING(def, event, emitter, "group", def->vxlan_params.group);
```

Comment:
IIUC Networkmanager uses a single setting to specify "remote" and "group", I wonder if we could do the same? Would the "remote" and "group" settings ever be used at the same time? Or can we deduct from the setting's value which one was meant by the user?

https://networkmanager.dev/docs/api/latest/nm-settings-nmcli.html#id-1.2.9.4.43

Reply:
Ineed, Group= will always be a multicast IP address (i.e. from a reserved IPv4/6 range), so we can implicitly derive this information, therefore I'd like to reduce those two fields to just a single `remote` field, that is than translated to Remote= or Group= for systemd-networkd depending on if it is a multicast IP or not.